### PR TITLE
fix: publish variables

### PR DESCRIPTION
**Pull request by lantoli**
## Proposed changes

_Jira ticket:_ [INTMDB-1185](https://jira.mongodb.org/browse/INTMDB-1185)

Fix publish variables in GH action

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run `make fmt` and formatted my code
- [X] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

**Review by maastha — APPROVED** [review]
Are LDAP tests run in git actions?

File: .github/workflows/cfn-publish-resource.yaml [hunk 1]
```diff
@@ -59,10 +59,10 @@ jobs:
       AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY_PUBLISHING}}
       AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID_PUBLISHING}}
 
-      # ldap-configuration
-      LDAP_BIND_PASSWORD: ${{ secrets.LDAP_BIND_PASSWORD }}
-      LDAP_HOST_NAME: ${{ secrets.LDAP_HOST_NAME }}
-      LDAP_BIND_USER_NAME: ${{ secrets.LDAP_BIND_USER_NAME }}
+      # LDAP config
```
[the remark this answers is not in the document]
Comment (lantoli) [reply .github/workflows/cfn-publish-resource.yaml:62]: no, this is for publishing

File: .github/workflows/cfn-publish-resource.yaml [hunk 2]
```diff
@@ -59,24 +59,24 @@ jobs:
...
+      WEBHOOK_UPDATE_SECRET: ${{ secrets.WEBHOOK_UPDATE_SECRET_PUBLISHING }}
+      PROMETHEUS_USER_NAME: ${{ secrets.PROMETHEUS_USER_NAME_PUBLISHING }}
+      PROMETHEUS_PASSWORD_NAME: ${{ secrets.PROMETHEUS_PASSWORD_NAME_PUBLISHING }}
+      PAGER_DUTY_CREATE_SERVICE_KEY: ${{ secrets.PAGER_DUTY_CREATE_SERVICE_KEY_PUBLISHING }}
+      PAGER_DUTY_UPDATE_SERVICE_KEY: ${{ secrets.PAGER_DUTY_UPDATE_SERVICE_KEY_PUBLISHING }}
+      DATA_DOG_CREATE_API_KEY: ${{ secrets.DATA_DOG_CREATE_API_KEY_PUBLISHING }}
+      DATA_DOG_UPDATE_API_KEY: ${{ secrets.DATA_DOG_UPDATE_API_KEY_PUBLISHING }}
+      OPS_GENIE_API_KEY: ${{ secrets.OPS_GENIE_API_KEY_PUBLISHING }}
+      MICROSOFT_TEAMS_WEBHOOK_CREATE_URL: ${{ secrets.MICROSOFT_TEAMS_WEBHOOK_CREATE_URL_PUBLISHING }}
+      MICROSOFT_TEAMS_WEBHOOK_UPDATE_URL: ${{ secrets.MICROSOFT_TEAMS_WEBHOOK_UPDATE_URL_PUBLISHING }}
```
Comment (marcosuma) [inline .github/workflows/cfn-publish-resource.yaml:79]: why did you add _PUBLISHING at the end?

Comment (lantoli) [reply .github/workflows/cfn-publish-resource.yaml:79]: I look at the repo secrets and this is the name of the secrets as they were created

Comment (lantoli) [reply .github/workflows/cfn-publish-resource.yaml:79]: they've used only for some resources.  It works fine for resource that don't need these variables, it'll mainly fail for ldap or third-party resources.

Comment (andreaangiolillo) [reply .github/workflows/cfn-publish-resource.yaml:79]: even if they did not use the github action, the publishing.sh script is used also by the old release process. With this change, the old release process will fail. I would update the secrets in the action and repo instead so we can still use the old release process if needed.

Comment (andreaangiolillo) [reply .github/workflows/cfn-publish-resource.yaml:79]: sorry, nvm my message, I see that you indeed added the secret to the repo. I got confused 😕
